### PR TITLE
Windows support for preserving the destination IP as the VIP when loadbalancing with DSR

### PR DESCRIPTION
**Pull request by ksubrmnn**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature
> /kind flake

**What this PR does / why we need it**:
This PR allows the user to specify the annotation "preserve-destination". When this is configured,  DNAT is skipped and loadbalanced packets are delivered with VIP/ClusterIP as the destination IP.

This PR also allows for local NodePort access from the host/local pods with the LocalRoutedVIP feature.

**Which issue(s) this PR fixes**:

Kube Proxy did not support preserving the destination IP.
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Windows support for preserving the destination IP as the VIP when loadbalancing with DSR.
```

### CODE REVIEW

**Comment by PatrickLang** [comment]
Today is code freeze for 1.14. Is 1.15 OK, and what's the OS version needed? I don't think it can make 1.14 since it's not already in the PR queue.

cc @michmike @craiglpeters

**Comment by ksubrmnn** [comment]
> Today is code freeze for 1.14. Is 1.15 OK, and what's the OS version needed? I don't think it can make 1.14 since it's not already in the PR queue.
> cc @michmike @craiglpeters

1.15 is fine. This PR still needs to be worked on/validated.

**Comment by ksubrmnn** [comment]
@kubernetes/dep-approvers

**Comment by ksubrmnn** [comment]
@dims can you PTAL at this vendoring change?

**Comment by dims** [comment]
/approve

(the vendoring changes)

**Review by dineshgovindasamy — APPROVED** [review]
/LGTM

**Comment by cblecker** [comment]
I'm good with the vendor changes here.

/assign @liggitt

File: pkg/proxy/winkernel/proxier.go [hunk 1]
```diff
@@ -204,6 +213,14 @@ func newServiceInfo(svcPortName proxy.ServicePortName, port *v1.ServicePort, ser
 	if service.Spec.SessionAffinity == v1.ServiceAffinityClientIP && service.Spec.SessionAffinityConfig != nil {
 		stickyMaxAgeSeconds = int(*service.Spec.SessionAffinityConfig.ClientIP.TimeoutSeconds)
 	}
+
+	klog.Infof("Service %q preserve-destination: %v", svcPortName.NamespacedName.String(), service.Annotations["preserve-destination"])
+
+	preserveDIP := service.Annotations["preserve-destination"] == "true"
```
Comment (liggitt) [inline pkg/proxy/winkernel/proxier.go]: cc @kubernetes/sig-network-api-reviews 

is this annotation-driven behavior novel for in-tree components?

**Comment by liggitt** [comment]
vendor bump looks fine.

one question on use of the annotations for driving this behavior. @thockin might have the most context on use of annotations in services like this

/assign @thockin

**Comment by PatrickLang** [comment]
just realized this was in review queue but not tagged
/milestone v1.15

File: pkg/proxy/winkernel/proxier.go (hunk 1 above)
Comment (madhanrm) [reply pkg/proxy/winkernel/proxier.go]: Currently this is specific to windows, but can be extendend to general kube-proxy and can be implemented across.

**Review by thockin — COMMENTED** [review]
Thanks!

/approve

**Pull request merged**